File: c++/lec1_jan27/isprime.cpp
#include <iostream>
using namespace std;

int main(int argc, char** argv){
	int n;
	int div = 3;

	cout << "Please enter a number :";
	cin >> n;

	if(n < 2 || (n % 2 == 0 && n!=2)) {
		cout << "Not prime" << endl;
		return 0;
	}

	while(div * div <= n){
		if(n % div == 0){
			cout << "Not prime" << endl;
			return 0;
		}
		div += 2;
	}
	cout << "Prime" << endl;
	return 0;
}

File: rust/lec1_jan27/is_prime.rs
use std::io;

fn main() {
	let mut n = String::new();

	println!("Please enter a number: ");

	io::stdin().read_line(&mut n)
		.expect("Error reading input");

	let n: u32 = n.trim().parse()
		.expect("Please enter an integer >= 1!");

	let mut div = 2;
	while div * div <= n {
		if n % div == 0 {
			println!("Not Prime");
			return;
		}
		div += 1;
	}
	println!("Prime");
}